packages/framework/client-logger/app-insights-logger/src/fluidAppInsightsLogger.ts

```diff
@@ -9,6 +9,64 @@ import {
 	type ITelemetryBaseLogger,
 } from "@fluidframework/core-interfaces";
 
+/**
+ * An Enum of Fluid telemetry event categories
+ */
+export const TelemetryEventCategory = {
+	PERFORMANCE: "performance",
+	GENERIC: "generic",
+	ERROR: "error",
+} as const;
+
+/**
+ * The type for the enum of Fluid telemetry event categories
+ */
+export type TelemetryEventCategory =
+	(typeof TelemetryEventCategory)[keyof typeof TelemetryEventCategory];
+
+/**
+ * The configuration object for the {@link FluidAppInsightsLogger}
+ */
+export interface FluidAppInsightsLoggerConfig {
+	filterConfig: {
+		/**
+		 * Determines whether all telemetry events are sent or not sent by default and whether filters will exclude matching telemetry events or include them.
+		 *
+		 * "inclusive" mode means all logs are NOT SENT by default and only the events that match specified filters will be sent (included).
+		 *
+		 * "exclusive" mode means all logs ARE SENT by default and only the events that match specified filters will be  not be sent (excluded).
+		 */
+		mode: "inclusive" | "exclusive";
+		/**
+		 * Controls the default filtering of log events by their category.
+		 * This can be overriden with namespace level filters
+		 */
+		filters?: TelemetryFilter[];
+	};
+}
+
+/**
+ * Object used with an {@link FluidAppInsightsLoggerConfig}
+ * to define logic for filtering of telemetry events
+ */
+export interface TelemetryFilter {
+	/**
+	 * The category {@link (TelemetryEventCategory:type)} of telemetry event that this filter applies to
+	 */
+	category: TelemetryEventCategory;
+	/**
+	 * The priority in which this filter should be evaluated.
+	 * A lower number means the filter will be evaluated earlier.
+	 *
+	 * If there are two filters with the same priority,
+	 * there is no guarentee to the order in which they will be evaluated
+	 */
+	priority: number;
```

Comment:
I'm not sure priority matters. It would if each filter could be inclusive/exclusive on its own, but if that config is global to all filters (and I wouldn't try to change that), then I think the order in which filters are applied won't produce different results...

Reply:
1. Exclusive means all events are sent and filters exclude matching events.
2. Inclusive means no events are sent and filters include matching events.

On the thought of priority, I think you are right I was thinking a bit to ahead here. I was planning to add exclusions but it doesn't come into play until the namespace PR and even then I can't find much use for priority in the current state.

I've updated the PR to remove priority.